Cargo.toml

```diff
@@ -17,6 +17,8 @@ categories = ["asynchronous", "cryptography", "network-programming"]
 [dependencies]
 native-tls = "0.2.3"
 futures-io = "0.3.1"
+thiserror = "1.0.9"
+futures-util = "0.3.1"
```

Comment:
Ah, I think using `async-std` with default features off and `std` enabled might be a better fit here; should bring the overall compile times down when used (:

Reply:
hmm, I was really hoping to keep it only dependent on `futures`